src/components/specimen/Specimen.tsx

```diff
@@ -0,0 +1,208 @@
+import FilterSpecimen from 'api/specimen/FilterSpecimen';
+import GetSpecimenDigitalMedia from 'api/specimen/GetSpecimenDigitalMedia';
+import GetSpecimenAnnotations from 'api/specimen/GetSpecimenAnnotations';
+import GetSpecimenVersions from 'api/specimen/GetSpecimenVersions';
+
+
+const Specimen = () => {
+    /* Configure Store */
+    const dispatch = useAppDispatch();
+
+    /* Hooks */
+    const params = useParams();
+
+    /* Base variables */
+    const specimen = useAppSelector(getSpecimen);
+    const specimenDigitalMedia = useAppSelector(getSpecimenDigitalMedia);
+    const specimenAnnotations = useAppSelector(getSpecimenAnnotations);
+    const annotateTarget = useAppSelector(getAnnotateTarget);
+    const [version, setVersion] = useState<number>();
+    const [versions, setVersions] = useState<number[]>([]);
+
+    /* Onload / Specimen or Version change: Check for Specimen, otherwise grab (specific version) from database */
+    useEffect(() => {
+        /* Fetch Specimen if not present or version has changed */
+        if (Object.keys(specimen).length === 0 || (version && specimen.version !== version)) {
+            /* Get Specimen */
+            GetSpecimen(`${params['prefix']}/${params['suffix']}`, version).then((specimen) => {
+                if (specimen) {
+                    /* Filter data */
+                    specimen.filtered = FilterSpecimen(specimen);
+
+                    dispatch(setSpecimen(specimen));
+
+                    /* Check Specimen Details */
+                    CheckDetails(specimen);
+                }
+            });
+        } else {
+            /* Check Specimen Details */
+            CheckDetails(specimen);
+        }
+    }, [specimen, version]);
+
+    /* Onchange of the Annotation Target's annotations: Check if changes occured */
+    useEffect(() => {
+        /* Check if the specimen annotations differ from the target annotations */
+        if (annotateTarget.annotations.length > 0 &&
+            JSON.stringify(specimenAnnotations[annotateTarget.property]) !== JSON.stringify(annotateTarget.annotations)) {
+            CheckAnnotations(specimen);
+        }
+    }, [annotateTarget.annotations]);
+
+    /* Function for checking Specimen Digital Media */
+    const CheckDigitalMedia = (specimen: SpecimenType) => {
+        GetSpecimenDigitalMedia(specimen.id).then((digitalMedia) => {
+            if (digitalMedia) {
+                dispatch(setSpecimenDigitalMedia(digitalMedia));
+            }
+        });
+    }
+
+    /* Function for checking Specimen Annotations */
+    const CheckAnnotations = (specimen: SpecimenType) => {
+        GetSpecimenAnnotations(specimen.id).then((annotations) => {
+            if (annotations) {
+                dispatch(setSpecimenAnnotations(annotations));
+            }
+        });
+    }
+
+    /* Function for a combined check on Specimen Digital Media, Annotations and Versions */
+    const CheckDetails = (specimen: SpecimenType) => {
```

Comment:
We should really make a single call out of this. At least for media and specimen as in the next version annotations will be behind a tab (so we call annotations when someone clicks that tab). I will make a userstory for this

Reply:
https://naturalis.atlassian.net/browse/DD-255